appiconloader-coil/src/main/java/me/zhanghai/android/appiconloader/coil/AppIconFetcher.java

```diff
@@ -21,52 +21,62 @@
 import android.graphics.Bitmap;
 import android.graphics.drawable.BitmapDrawable;
 
+import androidx.annotation.Px;
+
 import org.jetbrains.annotations.NotNull;
-import org.jetbrains.annotations.Nullable;
 
-import androidx.annotation.NonNull;
-import androidx.annotation.Px;
-import coil.bitmap.BitmapPool;
+import coil.ImageLoader;
 import coil.decode.DataSource;
-import coil.decode.Options;
 import coil.fetch.DrawableResult;
 import coil.fetch.FetchResult;
 import coil.fetch.Fetcher;
-import coil.size.Size;
+import coil.request.Options;
 import kotlin.coroutines.Continuation;
 import me.zhanghai.android.appiconloader.AppIconLoader;
 
-public class AppIconFetcher implements Fetcher<PackageInfo> {
-    @NonNull
+public class AppIconFetcher implements Fetcher {
+    @NotNull
     private final AppIconLoader mLoader;
-    @NonNull
+    @NotNull
     private final Context mContext;
+    @NotNull
+    private final PackageInfo mPackageInfo;
 
     public AppIconFetcher(@Px int iconSize, boolean shrinkNonAdaptiveIcons,
-                          @NonNull Context context) {
-        context = context.getApplicationContext();
-        mLoader = new AppIconLoader(iconSize, shrinkNonAdaptiveIcons, context);
+                          @NotNull Context context, @NotNull PackageInfo packageInfo) {
         mContext = context;
+        mLoader = new AppIconLoader(iconSize, shrinkNonAdaptiveIcons, context);
```

Comment:
`Fetcher`s are short-lived in Coil 2 so we shouldn't be creating a new `AppIconLoader` for every request - insetad, we should create it in the `Factory` and pass it to the `Fetcher`.